scripts/globals/znm.lua

```diff
@@ -57,86 +153,94 @@ xi.znm.soultrapper.getZeniValue = function(target, user, item)
     local zeni = 5
 
     -- HP% Component
-    local hpMultiplier = math.min(100 / hpp, 5)
-    if hpp <= 5 then
-        hpMultiplier = 10
-    end
-    zeni = zeni * hpMultiplier
-
-    -- In-Demand System Component
-    -- TODO: Make rotating server var
-    local inDemandSystem = xi.ecosystem.DRAGON
-    if system == inDemandSystem then
-        zeni = zeni * 1.5
-    end
-
-    -- NM/Rarity Component
-    if isNM then
-        zeni = zeni * 1.5
-    end
+    zeni = zeni * xi.znm.SOULPLATE_HPP_MULT * math.exp(100-hpp)
 
     -- Distance Component
-    zeni = zeni * utils.clamp((1 / distance) * 2, 1, 2)
+    zeni = zeni * utils.clamp((1 / distance) * 8, 1, 2)
 
     -- Angle/Facing Component
     if isFacing then
-        zeni = zeni * 1.5
+        zeni = zeni * xi.znm.SOULPLATE_FACING_MULT
     end
 
     -- Bonus for HS Soul Plate
     if user:getEquipID(xi.slot.AMMO) == xi.items.BLANK_HIGH_SPEED_SOUL_PLATE then
-        zeni = zeni * 1.5
+        zeni = zeni * xi.znm.SOULPLATE_HS_MULT
+    end
+
+    -- Check for Sanraku's "Subject of Interest"
+    if xi.znm.isCurrentInterest(target:getSuperFamily(), target:getFamily()) then
```

Comment:
You cant check check for fauna/interest when the picture is taken.
This must be done at trade in.

There are multiple reports of a player going out to take take pictures of mob X when it was the subOfInterest.
Then trading in 10 plates worth and having the zeni value drop somewhere in the middle of the 10 plates due to change of interest.